File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.kt
class Solution {
    fun findMin(nums: IntArray): Int {
        if(nums.size == 1) {
            return nums[0]
        }
        if(nums[0] < nums[nums.size-1]) {
            return nums[0]
        } 
        for(i in 0..nums.size-1) {
            if(nums[i] > nums[i+1]) {
                 return nums[i+1]
            }
        }        
        return 0
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> prifix(nums.size());
        vector<int> suffix(nums.size());
        vector<int> res(nums.size());
        prifix[0] = 1;
        suffix[nums.size()-1] = 1;
        for(int i=1;i<nums.size();i++) {
            prifix[i] = nums[i-1] * prifix[i-1];
        }

        for(int i=nums.size()-2;i>=0;i--) {
            suffix[i] = nums[i+1] * suffix[i+1];
        }

        for(int r=0;r<nums.size();r++) {
            res[r] = prifix[r] * suffix[r];       
        } 
        return res;
    }
};
